lib/jekyll/cache.rb

```diff
@@ -136,14 +150,16 @@ def key?(key)
     end
 
     def disk_cache_enabled?
-      !!Jekyll::Cache.disk_cache_enabled
+      !!(Jekyll::Cache.cache_dir && Jekyll::Cache.disk_cache_enabled)
     end
 
     private
 
     # Given a hashed key, return the path to where this item would be saved on disk.
     def path_to(hash = nil)
-      @base_dir ||= File.join(Jekyll::Cache.cache_dir, @name)
+      return unless disk_cache_enabled?
+
+      @base_dir ||= File.join(Jekyll::Cache.cache_dir, "Jekyll/Cache", @name)
```

Comment:
Can we use lower case here?
```suggestion
      @base_dir ||= File.join(Jekyll::Cache.cache_dir, "jekyll/cache", @name)
```

Reply:
Also, does this need to be a dot directory or anything like that?

Reply:
1. `Jekyll/Cache` was the path that Pat Hawks originally chose:
https://github.com/jekyll/jekyll/blob/7096885e98a72a3ed8cbe4c2c666d0f490211ea9/lib/jekyll/site.rb#L472

However, I don't see why we shouldn't opt for the downcased names.. perhaps, it was to avoid collision with existing directories created by third-party plugins..? Not sure.

2. There's no need for it to be a dot directory because the default value for `Jekyll::Cache.cache_dir` is `.jekyll-cache`. If a user overrides it, the onus is on them.